### DISCUSSION TITLE
Can i ignore parametres for test history matching?

### DISCUSSION THREAD

**Discussion by Elvintest** [allure-framework/allure2#2353] [Questions & Support] [answered]
Hello, we use java/junit5 test framework, and we have automatic rerun of failed tests, but due to specific conditions ecach launched test has such parameter 
```
"BSPlatform": "ForkJoinPool-3-worker-1"
``` 
after a test gets reruned it might have(most probably) a different value of  "BSPlatform" parameter, for example
```
"BSPlatform": "ForkJoinPool-2-worker-4"
``` 
For allure it seems a different test so it displayed several runs of one tests as different tests. Is there any option to ignore some specific parameter or parametres for matching test history without setting a value of this parameter?
Like we can't use a method 
```
public static <T> T parameter(String name, T value) {
    return parameter(name, value, (Boolean)null, (Parameter.Mode)null);
}
```
cause it needs a certain value of our parameter which we don't know before test gets run.

**Comment by baev** [comment, marked as answer]
You can use Allure lifecycle test listener to "fix" resulting json after test finishes. The other option would be to use 'Param' annotation (I assume you're using parameterization via test method arguments) and mark parameter as excluded (from history id) and hidden (from report)

**Reply by Elvintest** [reply]
Thank you, the first option has helped!
